lego/apps/users/views/users.py

```diff
@@ -184,19 +184,15 @@ def destroy(self, request, *args, **kwargs):
         serializer_class=PhotoConsentSerializer,
     )
     def update_photo_consent(self, request, *args, **kwargs):
-        serializer = self.get_serializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
         user = self.get_object()
 
         if not request.user.has_perm(EDIT, user):
             raise PermissionDenied(detail="Cannot update other user's consent")
 
-        PhotoConsent.objects.update_or_create(
-            user=user,
-            year=serializer.validated_data["year"],
-            semester=serializer.validated_data["semester"],
-            domain=serializer.validated_data["domain"],
-            defaults={"is_consenting": serializer.validated_data["is_consenting"]},
-        )
+        serializer = self.get_serializer(data={"user": user, **request.data})
```

Comment:
Any reason for changing from `data=request.data`? Is the user not in the data?

Reply:
No, as you can see, the user is being retrieved in the `get_object` method call further up. If you look at the old code, it was this user instance that was being user elsewhere as well.